client/src/modules/dashboard/Events/components/EventForm.tsx

```diff
@@ -42,14 +55,29 @@ const EventForm: React.FC<EventFormProps> = (props) => {
       capacity: data.capacity,
       start_at: new Date(data.start_at).toISOString().slice(0, 16),
       ends_at: new Date(data.ends_at).toISOString().slice(0, 16),
+      sponsors: data.sponsors,
       tags: (data.tags || []).map((t) => t.name).join(', '),
       venueId: data.venueId,
+      image_url: data.image_url,
     };
   }, []);
 
-  const { register, handleSubmit, watch, setValue } = useForm<EventFormData>({
-    defaultValues,
+  const { register, control, handleSubmit, watch, setValue, getValues } =
+    useForm<EventFormData>({
+      defaultValues,
+    });
+
+  const {
+    fields: sponsors,
+    append,
+    remove,
+  } = useFieldArray({
+    name: 'sponsors',
+    keyName: 'key',
+    control,
   });
+
```

Comment:
```suggestion
  const firstSponsorType = sponsorData?.sponsors[0]?.type as string | undefined;
  const firstSponsorId = sponsorData?.sponsors[0]?.id as number | undefined;
```
We can use these (see below) to make sure we always add a valid type of sponsor.  That way if a user just clicks `Add` once then `Save Event Changes` it should be valid.  It doesn't handle duplicate sponsors, but one step at a time!

Reply:
@ojeytonwilliams , Thank you! I've modified the code to default to first sponsor type and id.